### PR TITLE
initial commit for 30581

### PR DESCRIPTION
**Pull request by archanabpatil**
Signed-off-by: archanabpatil <66412855+archanabpatil@users.noreply.github.com>

# Problem Statement
-Added TA for test TEST-40745 and TEST-40993

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/s3/multipart_upload/test_multipart_getput.py [hunk 1]
```diff
@@ -829,3 +829,72 @@ def test_multipart_upload_test_40256(self):
...
+                                        "is_part_upload":True, "is_lst_complete_mpu":True})
+        process_put = multiprocessing.Process(target=self.s3_test_obj.put_object,
+                                              args=(self.bucket_name, self.object_name+str("put"),
+                                                    self.mp_obj_path))
+        process_mpu.start()
+        process_put.start()
+        process_mpu.join()
+        process_put.join()
+        res = self.s3_test_obj.object_list(self.bucket_name)
+        if (self.object_name and self.object_name+str("put")) not in res[1]:
```
Comment (apoorva-rao) [inline tests/s3/multipart_upload/test_multipart_getput.py:858]: The conditional expression will need to be corrected
```suggestion
        if (self.object_name not in res[1] or
            object_name_put not in res[1]):
```

File: tests/s3/multipart_upload/test_multipart_getput.py [hunk 2]
```diff
@@ -829,3 +829,72 @@ def test_multipart_upload_test_40256(self):
...
+        uploaded_parts, keys, s3_background_io = self.s3_mpu_test_obj.start_ios_get_precalc_parts(
+            mp_config, self.mp_obj_path, log_prefix="TEST-40745_s3bench_ios", duration="0h3m",
+            s3_test_lib_obj=self.s3_test_obj)
+        random.shuffle(keys)
+        process_mpu = multiprocessing.Process(target=self.initiate_multipart,
+                                args=(self.bucket_name, self.object_name),
+                                kwargs={"parts": uploaded_parts,
+                                        "is_part_upload":True, "is_lst_complete_mpu":True})
+        process_put = multiprocessing.Process(target=self.s3_test_obj.put_object,
+                                              args=(self.bucket_name, self.object_name+str("put"),
```
Comment (apoorva-rao) [inline tests/s3/multipart_upload/test_multipart_getput.py:851]: Let's store the name of the object being uploaded in line 851 in a variable and use that across the test instead

Eg. object_name_put = self.object_name + "_put"

**Review by apoorva-rao — CHANGES_REQUESTED** [review]
@archanabpatil I have added inline comments for some minor changes. Please check the same.

File: tests/s3/multipart_upload/test_multipart_getput.py (hunk 2 above)
Comment (archanabpatil) [reply tests/s3/multipart_upload/test_multipart_getput.py:851]: aggreed

**Review by seagate-sarang-sawant — APPROVED** [review]

File: tests/s3/multipart_upload/test_multipart_getput.py [hunk 3]
```diff
@@ -829,3 +829,73 @@ def test_multipart_upload_test_40256(self):
         s3_background_io.stop()
         s3_background_io.cleanup()
         self.log.info("ENDED: Test Simple and Multipart upload of an object")
+
+    @pytest.mark.tags('TEST-40745')
+    @pytest.mark.s3_ops
+    @CTFailOn(error_handler)
+    def test_multipart_upload_test_40745(self):
```
Comment (seagate-sarang-sawant) [inline tests/s3/multipart_upload/test_multipart_getput.py:836]: Are you skipping checksum verification for all I/Os. We need that for completeness of the tests.

Comment (archanabpatil) [reply tests/s3/multipart_upload/test_multipart_getput.py:836]: I will note it and add during TE phase in next sprint

**Review by apoorva-rao — APPROVED** [review]

**Pull request merged**